tests/auto/schema/schema_objectclass.c

```diff
@@ -17,65 +17,140 @@ Ensure(returns_true_on_successful_append_of_objectclass) {
 
     struct ldap_schema_t *schema = ldap_schema_new(ctx);
     LDAPObjectClass *objectclass = talloc_zero(ctx, LDAPObjectClass);
+    objectclass->oc_names = talloc_array(ctx, char*, 2);
+    objectclass->oc_names[0] = "test_objectclass_name";
+    objectclass->oc_names[1] = NULL;
+    objectclass->oc_oid = "test_objectclass_oid";
 
     bool result = ldap_schema_append_objectclass(schema, objectclass);
 
     assert_that(result, is_equal_to(true));
     assert_that(ldap_schema_object_classes(schema)[0], is_equal_to(objectclass));
+    assert_that(ldap_schema_get_objectclass_by_oid(schema, "test_objectclass_oid"), is_equal_to(objectclass));
+    assert_that(ldap_schema_get_objectclass_by_name(schema, "test_objectclass_name"), is_equal_to(objectclass));
 
     talloc_free(ctx);
 }
 
-Ensure(returns_true_on_successful_append_to_full_capacity_to_objectclasses) {
+Ensure(returns_false_on_null_schema_with_valid_objectclass) {
     TALLOC_CTX *ctx = talloc_new(NULL);
-
-    struct ldap_schema_t *schema = ldap_schema_new(ctx);
-    schema->object_classes_capacity = 1;
     LDAPObjectClass *objectclass = talloc_zero(ctx, LDAPObjectClass);
 
-    bool result = ldap_schema_append_objectclass(schema, objectclass);
+    bool result = ldap_schema_append_objectclass(NULL, objectclass);
 
-    assert_that(result, is_equal_to(true));
-    assert_that(ldap_schema_object_classes(schema)[0], is_equal_to(objectclass));
+    assert_that(result, is_equal_to(false));  // Assuming that the function returns true on a null schema
 
-    result = ldap_schema_append_objectclass(schema, objectclass);
+    talloc_free(ctx);
+}
 
-    assert_that(result, is_equal_to(true));
-    assert_that(ldap_schema_object_classes(schema)[1], is_equal_to(objectclass));
+Ensure(returns_false_on_null_objectclass_with_valid_schema) {
+    TALLOC_CTX *ctx = talloc_new(NULL);
+
+    struct ldap_schema_t *schema = ldap_schema_new(ctx);
+
+    bool result = ldap_schema_append_objectclass(schema, NULL);
+
+    assert_that(result, is_equal_to(false));  // Assuming that the function returns true on a null object class
 
     talloc_free(ctx);
 }
 
-Ensure(returns_false_on_null_schema_with_valid_objectclass) {
+Ensure(get_objectclass_with_valid_schema) {
```

Comment:
On the second thought consider removing `get_objectclass_with_valid_schema` completely. As an alternative we have to remove cycles and use `talloc` also we need to restructure and rename test.